examples/ibc/interchain_accounts_test.go

```diff
@@ -0,0 +1,314 @@
+package ibc
+
+import (
+	"context"
+	"encoding/json"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	"github.com/strangelove-ventures/ibctest"
+	"github.com/strangelove-ventures/ibctest/ibc"
+	"github.com/strangelove-ventures/ibctest/relayer"
+	"github.com/strangelove-ventures/ibctest/test"
+	"github.com/strangelove-ventures/ibctest/testreporter"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zaptest"
+)
+
+// TestInterchainAccounts is a test case that performs simulations and assertions around some basic
+// features and packet flows surrounding interchain accounts. See: https://github.com/cosmos/interchain-accounts-demo
+func TestInterchainAccounts(t *testing.T) {
+	if testing.Short() {
+		t.Skip()
+	}
+
+	t.Parallel()
+
+	client, network := ibctest.DockerSetup(t)
+
+	rep := testreporter.NewNopReporter()
+	eRep := rep.RelayerExecReporter(t)
+
+	ctx := context.Background()
+
+	// Get both chains
+	cf := ibctest.NewBuiltinChainFactory(zaptest.NewLogger(t), []*ibctest.ChainSpec{
+		{
+			Name: "icad",
+			ChainConfig: ibc.ChainConfig{
+				Images: []ibc.DockerImage{{Repository: "ghcr.io/cosmos/ibc-go-icad", Version: "latest"}},
+			},
+		},
+		{
+			Name: "icad",
+			ChainConfig: ibc.ChainConfig{
+				Images: []ibc.DockerImage{{Repository: "ghcr.io/cosmos/ibc-go-icad", Version: "latest"}},
+			},
+		},
+	})
+
+	chains, err := cf.Chains(t.Name())
+	require.NoError(t, err)
+
+	chain1, chain2 := chains[0], chains[1]
+
+	// Get a relayer instance
+	r := ibctest.NewBuiltinRelayerFactory(
+		ibc.CosmosRly,
+		zaptest.NewLogger(t),
+		relayer.RelayerOptionExtraStartFlags{Flags: []string{"-p", "events", "-b", "100"}},
+	).Build(t, client, network)
+
+	// Build the network; spin up the chains and configure the relayer
+	const pathName = "test-path"
+	const relayerName = "relayer"
+
+	ic := ibctest.NewInterchain().
+		AddChain(chain1).
+		AddChain(chain2).
+		AddRelayer(r, relayerName).
+		AddLink(ibctest.InterchainLink{
+			Chain1:  chain1,
+			Chain2:  chain2,
+			Relayer: r,
+			Path:    pathName,
+		})
+
+	require.NoError(t, ic.Build(ctx, eRep, ibctest.InterchainBuildOptions{
+		TestName:          t.Name(),
+		Client:            client,
+		NetworkID:         network,
+		SkipPathCreation:  true,
+		BlockDatabaseFile: ibctest.DefaultBlockDatabaseFilepath(),
+	}))
+
+	// Fund a user account on chain1 and chain2
+	const userFunds = int64(10_000_000_000)
+	users := ibctest.GetAndFundTestUsers(t, ctx, t.Name(), userFunds, chain1, chain2)
+	chain1User := users[0]
+	chain2User := users[1]
+
+	// Generate a new IBC path
+	err = r.GeneratePath(ctx, eRep, chain1.Config().ChainID, chain2.Config().ChainID, pathName)
```

Comment:
Is there a reason to do the path client/conns/channels steps explicitly in the test rather than letting the `Interchain` `Build` do it?

Reply:
I was skipping the automatic path initialization because we need to explicitly create an `ORDERED` channel. Also, in ICA the channel handshake is initiated by the user via the `intertx register` command which fires off `MsgRegisterInterchainAccount`